lib/api/utilities/grape_helper.rb

```diff
@@ -48,12 +48,13 @@ def grape_error_for(env, api)
         end
       end
 
-      def error_response(rescued_error, error = nil, rescue_subclasses: nil, headers: {})
+      def error_response(rescued_error, error = nil, rescue_subclasses: nil, headers: ->() { {} })
         default_response = lambda do |e|
           representer = ::API::V3::Errors::ErrorRepresenter.new e
+          resp_headers = instance_exec &headers
```

Comment:
I was struggling to fully grasp the reason behind the `instance_exec`. Just to confirm: You need to evaluate `env` (in [API::Root.auth_headers](https://github.com/machisuji/openproject/blob/feature/suppress_basic_auth_prompt/lib/api/root.rb#L147-L149) in the context of `GrapeHelper` ?